dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -5,6 +5,60 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   # GET /users/auth/:provider/callback
   def all
+    if should_connect_provider?
+      connect_provider
+    else
+      login
+    end
+  end
+
+  # GET /users/auth/:provider/connect
+  def connect
+    return head(:bad_request) unless current_user&.migrated? && params[:provider].present?
```

Comment:
:tada: This is awesome!  Do we want to check that the provider is in our list of supported providers (which might expand over time) or are we satisfied that the `omniauth_authorize_path` will do that?

Reply:
we do check that the provider is supported from our list in the callback (`connect_provider` method), but it's probably a good idea to do it here too